x-pack/plugins/enterprise_search/server/__mocks__/router.mock.ts

```diff
@@ -68,7 +68,14 @@ export class MockRouter {
   public validateRoute = (request: MockRouterRequest) => {
     if (!this.payload) throw new Error('Cannot validate wihout a payload type specified.');
 
-    const [config] = this.router[this.method].mock.calls[0];
+    // Ignoring a TS error here because it's tricky to resolve and unimportant to fix
+    // @ts-ignore
+    const configForMethodAndPath = this.router[this.method].mock.calls.find(
+      ([config]: Array<{ path: string }>) => config.path === this.path
+    );
+    if (!configForMethodAndPath)
+      throw new Error(`No route registered for ${this.method} & ${this.path}`);
```

Comment:
[not a blocker, I can open this up as a follow-up PR or you can cherry-pick this into your current PR, up to you]

@JasonStoltz I'd actually strongly rather pull out a single internal helper for this. Both `callRoute` and `validateRoute` are now reusing essentially the same find logic but written slightly differently, we can simplify this and remove the ts-ignore easily with this:

https://github.com/constancecchen/kibana/commit/b25f4d72414f77f0601d7c6604a0128386415e9d

Reply:
db7e587577029baf45e67d6e1358e40ee87f9ac9